src/java.sql.rowset/share/classes/javax/sql/rowset/serial/SQLOutputImpl.java

```diff
@@ -321,24 +321,20 @@ public void writeTimestamp(java.sql.Timestamp x) throws SQLException {
      *        use by a <code>SQLData</code> object attempting to write the attribute
      *        values of a UDT to the database.
      */
-    @SuppressWarnings("unchecked")
     public void writeCharacterStream(java.io.Reader x) throws SQLException {
-         BufferedReader bufReader = new BufferedReader(x);
-         try {
-             int i;
-             while( (i = bufReader.read()) != -1 ) {
+        BufferedReader bufReader = new BufferedReader(x);
+        try {
+            int i;
+            while ((i = bufReader.read()) != -1) {
```

Comment:
Is the single character read really necessary?
BufferedReader.readLine() returns null on EOF.
It seems plausable that only the readLine and writeString are needed in the loop.

Similarly, for the read loops below for InputStreams and binary stream.

Reply:
If this single character is `\n`, then direct replacing with `readLine` would change behavior of the method.

Reply:
I see your point and this PR isn't the place to figure out why a single newline in the attributes Vector would be significant vs an empty string or no entry.